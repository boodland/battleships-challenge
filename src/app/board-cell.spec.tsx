import { fireEvent, getByRole, render } from '@testing-library/react';

import BoardCell from './board-cell';

describe('BoardCell', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <BoardCell
        shotted={false}
        hasShip={false}
        showShip={false}
        onCellClick={() => null}
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should display no text if no shotted', () => {
    const { container } = render(
      <BoardCell
        shotted={false}
        hasShip={false}
        showShip={false}
        onCellClick={() => null}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should display missed text if shotted and has no ship', () => {
    const { container } = render(
      <BoardCell
        shotted={true}
        hasShip={false}
        showShip={false}
        onCellClick={() => null}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should display hitted text if shotted and has ship', () => {
    const { container } = render(
      <BoardCell
        shotted={true}
        hasShip={true}
        showShip={false}
        onCellClick={() => null}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should display ship if has ship and show ship', () => {
    const { container } = render(
      <BoardCell
        shotted={false}
        hasShip={true}
        showShip={true}
        onCellClick={() => null}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should call onCellClick when clicking the cell', () => {
    const onCellClickSpy = vi.fn();
    const { container } = render(
      <BoardCell
        shotted={false}
        hasShip={true}
        showShip={true}
        onCellClick={onCellClickSpy}
      />
    );
    const button = getByRole(container, 'button');
    fireEvent.click(button);
    expect(onCellClickSpy).toHaveBeenCalledOnce();
  });
});
