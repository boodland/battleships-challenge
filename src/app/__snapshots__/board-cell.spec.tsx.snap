// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BoardCell > should display hitted text if shotted and has ship 1`] = `
<div>
  <button
    class="_cell_436b5f _has-ship_436b5f  _shotted_436b5f"
  >
    X
  </button>
</div>
`;

exports[`BoardCell > should display missed text if shotted and has no ship 1`] = `
<div>
  <button
    class="_cell_436b5f   _shotted_436b5f"
  >
    O
  </button>
</div>
`;

exports[`BoardCell > should display no text if no shotted 1`] = `
<div>
  <button
    class="_cell_436b5f   "
  />
</div>
`;

exports[`BoardCell > should display ship if has ship and show ship 1`] = `
<div>
  <button
    class="_cell_436b5f _has-ship_436b5f _show-ship_436b5f "
  />
</div>
`;
